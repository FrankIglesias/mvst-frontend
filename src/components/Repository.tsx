interface RepositoryProps {
  name: string;
  description?: string | null;
  url: string;
}

/**
 * Component for showing details of a repository
 *
 * @component
 * @example
 * const name = 'react';
 * const description = 'A declarative, efficient, and flexible JavaScript library for building user interfaces.';
 * const url = 'https://github.com/react';
 * return (
 *  <Repository name={name} description={description} url={url} />
 * )
 * @param {string} name - Name of the repository
 * @param {string} description - Description of the repository
 * @param {string} url - URL of the repository
 * @returns {JSX.Element} - Rendered Repository component
 */
function Repository({ name, description, url }: RepositoryProps) {
  return (
    <div className="border-solid border-white border-2 mb-2 rounded p-4">
      <a href={url}>
        <h4>{name}</h4>
      </a>
      <p>{description}</p>
    </div>
  );
}

export default Repository;
