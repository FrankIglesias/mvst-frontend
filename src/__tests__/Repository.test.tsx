import { render } from '@testing-library/react';
import Repository from '../components/Repository';

describe('Repository', () => {
  it('renders without crashing', () => {
    const { container } = render(
      <Repository name="test" description="test" url="test" />
    );
    expect(container).toMatchSnapshot();
  });
});
