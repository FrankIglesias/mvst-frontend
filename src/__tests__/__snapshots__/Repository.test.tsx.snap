// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Repository > renders without crashing 1`] = `
<div>
  <div
    class="border-solid border-white border-2 mb-2 rounded p-4"
  >
    <a
      href="test"
    >
      <h4>
        test
      </h4>
    </a>
    <p>
      test
    </p>
  </div>
</div>
`;
